backend/src/main/java/org/cryptomator/hub/filters/VaultAdminOnlyFilterProvider.java

```diff
@@ -28,23 +33,47 @@ public class VaultAdminOnlyFilterProvider implements ContainerRequestFilter {
 	@Override
 	public void filter(ContainerRequestContext containerRequestContext) {
 		var vaultIdQueryParameter = getVaultIdQueryParameter(containerRequestContext);
-		var clientJwt = getClientJWT(containerRequestContext);
-		var unveridifedVaultId = getUnverifiedVaultId(clientJwt);
+		var clientJWT = getUnverifiedClientJWT(containerRequestContext);
+		var unveridifedVaultId = getUnverifiedVaultId(clientJWT);
 		if (vaultIdQueryParameter.equals(unveridifedVaultId)) {
 			var vault = Vault.<Vault>findByIdOptional(unveridifedVaultId).orElseThrow(NotFoundException::new);
 			var algorithm = Algorithm.ECDSA384(decodePublicKey(vault.authenticationPublicKey), null);
-			try {
-				JWT.require(algorithm).build().verify(clientJwt);
-			} catch (TokenExpiredException e) {
-				throw new VaultAdminTokenExpiredException("Token of Client-JWT expired");
-			} catch (JWTVerificationException e) {
-				throw new VaultAdminValidationFailedException("Different key used to sign the Client-JWT");
-			}
+			verify(buildVerification(algorithm), clientJWT);
 		} else {
 			throw new VaultAdminValidationFailedException("Other vaultId provided");
 		}
 	}
 
+	//visible for testing
+	void verify(JWTVerifier verifier, DecodedJWT clientJWT) {
+		try {
+			verifier.verify(clientJWT);
+		} catch (TokenExpiredException e) {
+			throw new VaultAdminTokenExpiredException("Token of Client-JWT expired");
+		} catch (IncorrectClaimException e) {
+			if (e.getClaimName().equals(RegisteredClaims.ISSUED_AT) || e.getClaimName().equals(RegisteredClaims.NOT_BEFORE)) {
+				throw new VaultAdminTokenNotYetValidException("Token of Client-JWT not yet valid");
+			} else {
+				throw new VaultAdminValidationFailedException("Incorrect claim exception");
+			}
+		} catch (JWTVerificationException e) {
+			throw new VaultAdminValidationFailedException("Different key used to sign the Client-JWT");
+		}
+	}
+
+	//visible for testing
+	JWTVerifier buildVerification(Algorithm algorithm) {
+		return buildVerifier(algorithm).build();
+	}
+
+	//visible for testing
+	Verification buildVerifier(Algorithm algorithm) {
+		return JWT.require(algorithm) //
+				.withClaim(RegisteredClaims.ISSUED_AT, (claim, jwt) -> jwt.getIssuedAt() != null) //
+				.withClaim(RegisteredClaims.NOT_BEFORE, (claim, jwt) -> jwt.getNotBefore() != null) //
+				.withClaim(RegisteredClaims.EXPIRES_AT, (claim, jwt) -> jwt.getExpiresAt() != null);
+	}
```

Comment:
method names vs. return types is still mixed up

Reply:
Fixed in https://github.com/cryptomator/hub/pull/133/commits/75da9109a510ef34ffaaf0d1bf0f6d80c21a9ad1